prov/efa/test/efa_unit_test_domain.c

```diff
@@ -0,0 +1,150 @@
+                                resource,
+                                FI_EFA_MR_ATTR_VALIDITY_RECV_PCI_BUS_ID,
+                                0,
+                                0 /* ignored */,
+                                0 /* ignored */);
+}
+
+/* run mr query for rdma read pci bus id */
+void test_efa_domain_open_ops_mr_query_rdma_read(struct efa_resource **state)
+{
+    struct efa_resource *resource = *state;
+
+    efa_unit_test_resource_construct(resource, FI_EP_RDM);
+
+    /* set rdma_read_pci_bus_id as 1 */
+    g_efa_unit_test_mocks.efadv_query_mr = &efa_mock_efadv_query_mr_rdma_read_pci_bus_id_1;
+
+    test_efa_domain_open_ops_mr_query_common(
+                                resource,
+                                FI_EFA_MR_ATTR_VALIDITY_RDMA_READ_PCI_BUS_ID,
+                                0 /* ignored */,
+                                1,
+                                0 /* ignored */);
+}
+
+/* run mr query for rdma recv pci bus id */
+void test_efa_domain_open_ops_mr_query_rdma_recv(struct efa_resource **state)
+{
+    struct efa_resource *resource = *state;
+
+    efa_unit_test_resource_construct(resource, FI_EP_RDM);
+
+    /* set rdma_recv_pci_bus_id as 2 */
+    g_efa_unit_test_mocks.efadv_query_mr = &efa_mock_efadv_query_mr_rdma_recv_pci_bus_id_2;
+
+    test_efa_domain_open_ops_mr_query_common(
+                                resource,
+                                FI_EFA_MR_ATTR_VALIDITY_RDMA_RECV_PCI_BUS_ID,
+                                0 /* ignored */,
+                                0 /* ignored */,
+                                2);
+}
+
+/* run mr query for both recv and rdma read's pci bus id */
+void test_efa_domain_open_ops_mr_query_recv_and_rdma_read(struct efa_resource **state)
+{
+    struct efa_resource *resource = *state;
+
+    efa_unit_test_resource_construct(resource, FI_EP_RDM);
+
+    /* set recv_pci_bus_id as 0, rdma_read_pci_bus_id as 1  */
+    g_efa_unit_test_mocks.efadv_query_mr = &efa_mock_efadv_query_mr_recv_and_rdma_read_pci_bus_id_0_1;
+
+    test_efa_domain_open_ops_mr_query_common(
+                                resource,
+                                FI_EFA_MR_ATTR_VALIDITY_RECV_PCI_BUS_ID | FI_EFA_MR_ATTR_VALIDITY_RDMA_READ_PCI_BUS_ID,
+                                0,
+                                1,
+                                0 /* ignored */);
+}
+
+#else
+
+void test_efa_domain_open_ops_mr_query_recv(struct efa_resource **state)
+{
+    skip();
+}
+
+/* run mr query for rdma read pci bus id */
+void test_efa_domain_open_ops_mr_query_rdma_read(struct efa_resource **state)
+{
+    skip();
+}
+
+/* run mr query for rdma recv pci bus id */
+void test_efa_domain_open_ops_mr_query_rdma_recv(struct efa_resource **state)
+{
+    skip();
+}
+
+/* run mr query for both recv and rdma read's pci bus id */
+void test_efa_domain_open_ops_mr_query_recv_and_rdma_read(struct efa_resource **state)
+{
+    skip();
+}
+
+#endif /* HAVE_EFADV_QUERY_MR */
```

Comment:
Is this the general pattern we follow with our unit tests? We are mocking out the implementation of our dependencies so we can unit test a particular interface. Why skip these tests if configure does not find the capability in rdma-core?

Reply:
@rajachan To mock the rdma-core verbs we need the symbols of `efadv_query_mr` `efadv_mr_attr` and `EFADV**`, which aren't available when they are not found in rdma-core headers? How can we not skip them in this case?

Reply:
>Is this the general pattern we follow with our unit tests?

Yes, we have done the same pattern to unit test something that is only introduced in newer rdma-core

Reply:
> In other words, I would expect to have a test here and see an expected error ENOSYS, rather than skipping.

I get this point, will add it

Reply:
That's the whole point of a mock. What behavior do you expect apps to see when they call into the open_ops API with the EFA provider built against a version of rdma-core that does not have that definition? ENOSYS? Then we need a test to check that behavior.

Reply:
Updated